aea/agent.py

```diff
@@ -252,46 +297,6 @@ def stop(self) -> None:
         """
         self.runtime.stop()
 
-    @abstractmethod
-    def setup(self) -> None:
-        """
-        Set up the agent.
-
-        :return: None
-        """
-
-    @abstractmethod
-    def act(self) -> None:
```

Comment:
Act should also be declared here with raise NotImplementedError

Reply:
it's redundant, it's not possible to construct an instance with abstract method, so main idea is to make abstract or raise not implemented error. not both